src/Symmetric.jl

```diff
@@ -0,0 +1,217 @@
+````math
+x_{new} = \frac{1}{2} \left( x_{old} + x_{old}^{\mathrm{T}} \right).$
+````
+The value of the variable `x` is overwritten.
+"""
+project_point!(M::SymmetricMatrices, x) = (x .= 1/2 * (x + transpose(x)))
+
+@doc doc"""
+    project_tangent!(M::SymmetricMatrices, w, x, v)
+    
+Implements the [`project_tangent!`](@ref project_tangent!(M::Manifold, w, x, v)) of the matrix `v` onto the tangent space of the manifold of [`SymmetricMatrices`](@ref),
+i.e. the [`SymmetricMatrices`](@ref) itself. The result is stored in `w`:
+````math
+w = \frac{1}{2} \left( v + v^{\mathrm{T}} \right)
+````
+
+"""
+project_tangent!(M::SymmetricMatrices, w, x, v) = (w .= 1/2 * (v + transpose(v)))
+
+
+@doc doc"""
+    inner(M::SymmetricMatrices, x, w, v)
+
+compute the inner product of the two tangent vectors `w,v` from the tangent
+plane at `x` on the [`SymmetricMatrices`](@ref) `M` using the restriction of the
+metric from the embedding, i.e. 
+````math
+(v,w)_x = \operatorname{tr}(v^{\mathrm{T}}w)
+````
+"""
+@inline inner(M::SymmetricMatrices, x, w, v) = dot(w, v)
+
+
+norm(M::SymmetricMatrices, x, v) = norm(v)
+
+"""
+    distance(M::SymmetricMatrices, x, y)
+
+by using the inherited metric, i.e. taking the Frobenius-norm of the difference.
+
+"""
+
+distance(M::SymmetricMatrices, x, y) = norm(x-y)
+
+@doc doc"""
+    exp!(M::SymmetricMatrices, y, x, v)
+Computes the exponential map [`exp!`](@ref exp!(M::Manifold, y, x, v, t=1)) of the tangent vector `v` at
+manifold point `x` lying on the [`SymmetricMatrices`](@ref) manifold `M`. 
+The result is saved in `y`. 
+For the [`SymmetricMatrices`](@ref), the exponential map is simply the sum, i.e. 
+````math
+    y = x + v
+````
+"""
+function exp!(M::SymmetricMatrices, y, x, v)
+    y .= x + v
+    return y
+end
+
+@doc doc"""
+    log!(M::SymmetricMatrices, v, x, y)
+Computes the logarithmic map from the manifold point `x` to the manifold point `y`, both lying on the [`SymmetricMatrices`](@ref) manifold `M`. 
+The result is saved in `v`. 
+For the [`SymmetricMatrices`](@ref), the logarithmic map is simply the difference, i.e. 
+````math
+    v = y-x
+````
+"""
+function log!(M::SymmetricMatrices, v, x, y)
+    v .= y-x
+    return v
+end
+
+function zero_tangent_vector!(M::SymmetricMatrices, v, x)
+    fill!(v, 0)
+    return v
+end
+
+
+@doc doc"""
+    vector_transport_to!(M::SymmetricMatrices, vto, x, v, y)
+Computes the [`vector_transport_to!`](@ref vector_transport_to!(M::Manifold, vto, x, v, y, ProjectionTransport())) 
+of the tangent vector `v` from the manifold point `x` on the [`SymmetricMatrices`](@ref) `M`
+to the manifold point $y\in M$. The result is stored in `vto`.
+Since the metric is inherited from the embedding space, this is just the identity.
+"""
+function vector_transport_to!(M::SymmetricMatrices, vto, x, v, y, ::ParallelTransport)
+    vto .= v
```

Comment:
Here you can just use `copyto!`, which will do no copy if `v` and `vto` are the same object.